File: server/src/main/java/dev/slimevr/config/DriftCompensationConfig.java
package dev.slimevr.config;

import dev.slimevr.Main;
import dev.slimevr.tracking.trackers.IMUTracker;
import dev.slimevr.tracking.trackers.TrackerJava;


public class DriftCompensationConfig {

	// Is drift compensation enabled
	private boolean enabled = false;

	// Amount of drift compensation applied
	private float amount = 0.8f;

	// Max resets for the calculated average drift
	private int maxResets = 6;

	public DriftCompensationConfig() {
	}

	public void updateTrackersDriftCompensation() {
		for (TrackerJava t : Main.getVrServer().getAllTrackers()) {
			TrackerJava tracker = t.get();
			if (tracker instanceof IMUTracker imuTracker) {
				imuTracker
					.setDriftCompensationSettings(
						getEnabled(),
						getAmount(),
						getMaxResets()
					);
			}
		}
	}

	public boolean getEnabled() {
		return enabled;
	}

	public void setEnabled(boolean enabled) {
		this.enabled = enabled;
	}

	public float getAmount() {
		return amount;
	}

	public void setAmount(float amount) {
		this.amount = amount;
	}

	public int getMaxResets() {
		return maxResets;
	}

	public void setMaxResets(int maxResets) {
		this.maxResets = maxResets;
	}
}

File: server/src/main/java/dev/slimevr/config/DriftCompensationConfig.kt
package dev.slimevr.config

import dev.slimevr.vrServer

class DriftCompensationConfig {

	// Is drift compensation enabled
	var enabled = true // TODO

	// Amount of drift compensation applied
	var amount = 0.8f // TODO

	// Max resets for the calculated average drift
	var maxResets = 6 // TODO
	fun updateTrackersDriftCompensation() {
		for (t in vrServer.allTrackers) {
			if (t.allowFiltering) {
				t.resetsHandler.readDriftCompensationConfig(this)
			}
		}
	}
}
